packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -34,12 +45,32 @@ class WindowsVersionValidator extends DoctorValidator {
 
     // Use the string split method to extract the major version
     // and check against the [kUnsupportedVersions] list
-    final ValidationType windowsVersionStatus;
-    final String statusInfo;
+    ValidationType windowsVersionStatus;
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    String statusInfo;
     if (matches.length == 1 &&
         !kUnsupportedVersions.contains(matches.elementAt(0).group(1))) {
       windowsVersionStatus = ValidationType.success;
       statusInfo = 'Installed version of Windows is version 10 or higher';
+
+      // Check if the Topaz OFD security module is running, and warn the user if it is.
+      // See https://github.com/flutter/flutter/issues/121366
+      final ProcessResult getProcessesResult = await _processLister.getProcessesWithPath();
+      if (getProcessesResult.exitCode != 0) {
+        windowsVersionStatus = ValidationType.partial;
+        statusInfo = 'Failed to execute Get-Process';
+        messages.add(ValidationMessage.error('Get-Process returned non-success exit code ${getProcessesResult.exitCode}'));
+      } else {
+        final RegExp topazRegex = RegExp(kCoreProcessPattern, caseSensitive: false,  multiLine: true);
+        final String processes = getProcessesResult.stdout as String;
+        final bool topazFound = topazRegex.hasMatch(processes);
+        if (topazFound) {
+          windowsVersionStatus = ValidationType.partial;
+          messages.add(ValidationMessage(statusInfo));
+          statusInfo = 'Topaz OFD may be running';
```

Comment:
Is this logic correct? Don't we want to set the `statusInfo` before we add it to the messages list?

Reply:
As you commented below, the purpose here is to create a validation message containing the info regarding the installed Windows version, as that was `statusInfo` before the Topaz related issue was discovered.